cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/preview/DefaultPreviewRunner.java

```
@@ -132,14 +144,19 @@
     this.levelDBTableService = levelDBTableService;
     this.structuredTableAdmin = structuredTableAdmin;
     this.structuredTableRegistry = structuredTableRegistry;
+    this.previewIdDirPath = Paths.get(cConf.get(Constants.CFG_LOCAL_DATA_DIR), "previewid").toAbsolutePath();
+    try {
+      Files.createDirectories(previewIdDirPath);
```

Should be done in the startup method and let the exception bubble out. Generally not good to throw exception from constructor for classes that are created via Guice injection.

---

make sense! moved to startup.